backend/src/plugins/seed/server/seed/content-types.js

```diff
@@ -24,24 +24,53 @@ module.exports = async (strapi) => {
       strapi.log.info("🌱 Menus already exist, skipping");
     } else {
       strapi.log.info("🌱 Create menus..");
+      let images;
       for (let i = 0; i < menus.length; i++) {
-        let data = menus[i];
-        data.items = data.items.map((item) => {
-          if (item.__component === "menu.submenu") {
-            const mappedSubmenuId = idMap.menu[item.submenu];
-            if (mappedSubmenuId == null) {
-              throw new Error(
-                `⚠️ Cannot find submenu with id ${item.submenu.id}`
-              );
+        let menu = menus[i];
+        images = [];
+        menu.items = menu.items.map((item) => {
+          switch (item.__component) {
+            case "menu.submenu": {
+              const mappedSubmenuId = idMap.menu[item.submenu];
```

Comment:
I see that `idMap` is initialized as an empty object at  https://github.com/iFixit/react-commerce/blob/ab1800e851a22afe3779c6a11c4d377f592356cb/backend/src/plugins/seed/server/seed/content-types.js#L13-L18 but the only place I see that `idMap.menu` gets populated is later on in the script at https://github.com/iFixit/react-commerce/blob/ab1800e851a22afe3779c6a11c4d377f592356cb/backend/src/plugins/seed/server/seed/content-types.js#L52-L56 does this mean the error will be thrown?

Reply:
This throws only if associated items do not come first in the seed.. For example, in order to create a menu, you need to create its submenus first. 

I don't love how seeds are managed, and I'd like to improve eventually, but for now this gets the job done